connectors/src/connectors/slack/bot.ts

```diff
@@ -264,9 +265,13 @@ async function botAnswerMessage(
       }
     }
   }
+
+  // Remove markdown to extract mentions.
+  const messageWithoutMarkdown = removeMarkdown(message);
+
   // Extract all ~mentions.
   const mentionCandidates =
-    message.match(/(?<!\S)~(?!~)([a-zA-Z0-9_-]{1,20})(?!\S)/g) || [];
+    messageWithoutMarkdown.match(/(?<!\S)~([a-zA-Z0-9_-]{1,20})(?!\S)/g) || [];
```

Comment:
I removed the negative look ahead to handle strikethrough since it will be stripped when removing markdown.